python/python/lance/dataset.py

```diff
@@ -1667,6 +1668,7 @@ def index_stats(self, index_name: str) -> Dict[str, Any]:
         index_stats = json.loads(self._ds.index_statistics(index_name))
         index_stats["num_indexed_rows"] = self._ds.count_indexed_rows(index_name)
         index_stats["num_unindexed_rows"] = self._ds.count_unindexed_rows(index_name)
+        index_stats["index_cache_size"] = self._ds.index_cache_size()
```

Comment:
There's some confusing naming here. Sometimes "index_cache_size" is the capacity of the index cache and sometimes it's the current number of entries. IIUC, the first meaning is already in use, so can we come up with a different name for the statistic? Maybe `index_cache_entry_count` or something?

Reply:
`index_cache_entry_count` sounds ok. Changed.